flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/split/KafkaPartitionSplit.java

```diff
@@ -35,17 +35,18 @@
 public class KafkaPartitionSplit implements SourceSplit {
     public static final long NO_STOPPING_OFFSET = Long.MIN_VALUE;
     // Indicating the split should consume from the latest.
-    public static final long LATEST_OFFSET = -1;
+    // @deprecated Only be used for compatibility with the history state, see FLINK-28303
+    @Deprecated public static final long LATEST_OFFSET = -1;
```

Comment:
Simiarly, we could remove it from Javadocs in `ReaderHandledOffsetsInitializer` (the javadocs describes how it supports latest which we want to avoid using).

Reply:
I was negligent. Fixed.